rasterio/_base.pyx

```diff
@@ -449,6 +450,48 @@ cdef class DatasetBase(object):
                     GDALGetRasterUnitType(self.band(j)) for j in self.indexes)
             return self._units
 
+    def block(self, bidx, i, j):
+        """Returns info about a particular block
+
+        Parameters
+        ----------
+        bidx: int
+            Band index, starting with 1.
+        i: int
+            Row index of the block, starting with 0.
+        j: int
+            Column index of the block, starting with 0.
+
+        Returns
+        -------
+        BlockInfo
+        """
+        cdef GDALMajorObjectH obj = NULL
+        cdef char *value = NULL
+        cdef const char *key_c = NULL
+
+        obj = self.band(bidx)
+
+        if self.driver == 'GTiff':
+            key_b = 'BLOCK_SIZE_{0}_{1}'.format(j, i).encode('utf-8')
+            key_c = key_b
+            value = GDALGetMetadataItem(obj, key_c, 'TIFF')
+            if value == NULL:
+                raise TIFFTagError(
+                    "Block i={0}, j={1} size can't be determined".format(i, j))
+            else:
+                size = int(value)
+        else:
+            size = None
```

Comment:
I think I'd prefer not to warn because there's no way for a user to change usage in a way that would avoid a warning.